src/inverted-index.js

```diff
@@ -0,0 +1,140 @@
+const utils = require('./utils.js');
+
+/**
+ * A class for Creating and searching an inverted index
+ */
+class Index {
+
+  /**
+   * [constructor Creates Object variables to hold
+   * 1. [jsonDatabase] Uploaded files
+   * 2. [indexFile] the indexed filenames and their contents
+   * 3. [searchResult] the search results
+   */
+  constructor() {
+    this.jsonDatabase = {};
+    this.indexFile = {};
+    this.searchResult = {};
+    this.indexView = '';
+  }
+
+  /**
+   * [saveUploads creates a key and value object item that stores the uploaded file(s)]
+   * @param  {string} fileName [filename]
+   * @param  {object} jsonFile [content of uploaded json file]
+   * @return {boolean} [returns true on succesful addition of object to datatbase]
+   */
+  saveUploads(fileName, jsonFile) {
+    if (!utils.isValid(fileName, jsonFile)) {
+      return false;
+    }
+    if (typeof jsonFile === 'string') {
+      jsonFile = JSON.parse(jsonFile);
+    }
+    this.jsonDatabase[fileName] = [];
+    Object.keys(jsonFile).forEach((keys) => {
+      this.jsonDatabase[fileName].push(jsonFile[keys]);
+    });
+    return true;
+  }
+
+  /**
+   * [getJsonDatabase function to return the saved uploads]
+   * @return {object} [the saved uploads]
+   */
+  getJsonDatabase() {
+    return this.jsonDatabase;
+  }
+
+  /**
+   * [createIndex Creates an index of the words in the received json file]
+   * @param  {string}   filePath [the key(filename) of the json value to index]
+   * @param  {Function} cb  [call back to return the indexed file object/an html format index table]
+   * @return {array} [an arrray of the indexed file result and the html Div of the index]
+   */
+  createIndex(filePath) {
+    const indexFile = this.indexFile;
+    const jsonDoc = this.jsonDatabase[filePath];
+    let joinedValues = '';
+    let wordArray = [];
+    if (indexFile[filePath]) {
+      return false;
+    }
+    indexFile[filePath] = {};
+    jsonDoc.forEach((element, index) => {
+      joinedValues = utils.cleanString((`${element.title} ${element.text}`));
+      wordArray = new Set(joinedValues.split(' '));
+      wordArray.forEach((word) => {
+        indexFile[filePath][word] = indexFile[filePath][word] || [];
+        indexFile[filePath][word].push(index);
+      });
+    });
+    this.indexFile = indexFile;
```

Comment:
Yeah, overdoing was because this.indexFile showed some inconsistent behavior within the forEach statement, so I fell back on this after trying a couple of solutions.